New Resource: `azurerm_workloads_sap_three_tier_virtual_instance` (#24384)
hashicorp/terraform-provider-azurerm: website/docs/r/workloads_sap_three_tier_virtual_instance.html.markdown

        size_in_gb      = 128
        sku_name        = "Premium_LRS"
      }

      disk_volume_configuration {
        volume_name     = "hana/shared"
        number_of_disks = 1
        size_in_gb      = 256
        sku_name        = "Premium_LRS"
      }

      disk_volume_configuration {
        volume_name     = "usr/sap"
        number_of_disks = 1
        size_in_gb      = 128
        sku_name        = "Premium_LRS"
      }

      disk_volume_configuration {
        volume_name     = "backup"
        number_of_disks = 2
        size_in_gb      = 256
        sku_name        = "StandardSSD_LRS"
      }

      disk_volume_configuration {
        volume_name     = "os"
        number_of_disks = 1
        size_in_gb      = 64
        sku_name        = "StandardSSD_LRS"
      }
    }

    resource_names {
      application_server {
        availability_set_name = "appAvSet"

        virtual_machine {
          host_name               = "apphostName0"
          os_disk_name            = "app0osdisk"
          virtual_machine_name    = "appvm0"
          network_interface_names = ["appnic0"]

          data_disk_names = {
            default = "app0disk0"

>           }

Reviewer: could we update this to reflect the new name/layout of this property?

        }
      }

      central_server {
        availability_set_name = "csAvSet"

        load_balancer {
          name                            = "ascslb"
          backend_pool_names              = ["ascsBackendPool"]
          frontend_ip_configuration_names = ["ascsip0"]
          health_probe_names              = ["ascsHealthProbe"]
        }

        virtual_machine {
          host_name               = "ascshostName"
          os_disk_name            = "ascsosdisk"
          virtual_machine_name    = "ascsvm"
          network_interface_names = ["ascsnic"]

          data_disk_names = {
            default = "ascsdisk"

>           }

Reviewer: could we update this to reflect the new name/layout of this property?

Author: Updated

        }
      }

      database_server {
        availability_set_name = "dbAvSet"

        load_balancer {
          name                            = "dblb"
          backend_pool_names              = ["dbBackendPool"]
          frontend_ip_configuration_names = ["dbip"]
          health_probe_names              = ["dbHealthProbe"]
        }

        virtual_machine {
          host_name               = "dbprhost"
          os_disk_name            = "dbprosdisk"
          virtual_machine_name    = "dbvmpr"
          network_interface_names = ["dbprnic"]

          data_disk_names = {
            hanaData   = "hanadatapr0,hanadatapr1"
            hanaLog    = "hanalogpr0,hanalogpr1,hanalogpr2"
            usrSap     = "usrsappr0"

>             hanaShared = "hanasharedpr0,hanasharedpr1"

Reviewer: could we update this to reflect the new name/layout of this property?

Author: Updated

* `publisher` - (Required) The publisher of the Image. Possible values are `RedHat` and `SUSE`. Changing this forces a new resource to be created.

* `sku` - (Required) The SKU of the Image. Changing this forces a new resource to be created.

* `version` - (Required) Specifies the version of the platform image or marketplace image used to create the virtual machine. Changing this forces a new resource to be created.

---

An `os_profile` block supports the following:

* `admin_username` - (Required) The name of the administrator account. Changing this forces a new resource to be created.

* `ssh_private_key` - (Required) The SSH public key that is used to authenticate with the Virtual Machine. Changing this forces a new resource to be created.

* `ssh_public_key` - (Required) The SSH private key that is used to authenticate with the Virtual Machine. Changing this forces a new resource to be created.

---

A `three_tier_configuration` block supports the following:

* `app_resource_group_name` - (Required) The name of the application Resource Group where SAP system resources will be deployed. Changing this forces a new resource to be created.

> ~> **Note:** While creating an SAP Three Tier Virtual Instance, the service would provision the extra SAP system/component in the `app_resource_group_name` that isn't defined in the HCL Configurations. At this time, if the `app_resource_group_name` is different from the Resource Group where SAP Three Tier Virtual Instance exists, you can set `prevent_deletion_if_contains_resources` to `false` to delete all resources defined in the HCL Configurations and the resources created in the `app_resource_group_name` with `terraform destroy`. However, if the `app_resource_group_name` is the same with the Resource Group where SAP Three Tier Virtual Instance exists, some resources such as the subnet defined in the HCL Configurations cannot be deleted with `terraform destroy` since the resources defined in the HCL Configurations are being referenced by the SAP system/component. In this case, you have to manually delete the SAP system/component before deleting the resources in the HCL Configurations.

Reviewer: ```suggestion
~> **Note:** While creating an SAP Three Tier Virtual Instance, the service will provision the extra SAP systems/components in the `app_resource_group_name` that are not defined in the HCL Configuration. At this time, if the `app_resource_group_name` is different from the Resource Group where SAP Three Tier Virtual Instance exists, you can set `prevent_deletion_if_contains_resources` to `false` to delete all resources defined in the HCL Configurations and the resources created in the `app_resource_group_name` with `terraform destroy`. However, if the `app_resource_group_name` is the same with the Resource Group where SAP Three Tier Virtual Instance exists, some resources, such as the subnet defined in the HCL Configuration, cannot be deleted with `terraform destroy` since the resources defined in the HCL Configuration are being referenced by the SAP system/component. In this case, you have to manually delete the SAP system/component before deleting the resources in the HCL Configuration.
```

Author: Updated

* `central_server_configuration` - (Required) A `central_server_configuration` block as defined below. Changing this forces a new resource to be created.

* `database_server_configuration` - (Required) A `database_server_configuration` block as defined below. Changing this forces a new resource to be created.

* `resource_names` - (Optional) A `resource_names` block as defined below. Changing this forces a new resource to be created.

* `high_availability_type` - (Optional) The high availability type for the three tier configuration. Possible values are `AvailabilitySet` and `AvailabilityZone`. Changing this forces a new resource to be created.

* `secondary_ip_enabled` - (Optional) Specifies whether a secondary IP address should be added to the network interface on all VMs of the SAP system being deployed. Defaults to `false`. Changing this forces a new resource to be created.

* `transport_create_and_mount` - (Optional) A `transport_create_and_mount` block as defined below. Changing this forces a new resource to be created.

~> **Note:** The file share configuration uses `skip` by default when `transport_create_and_mount` isn't set.

> ~> **Note:** Currently, the last segment of the Storage File Share resource manager ID in Swagger is defined as `/shares/` but it's unexpected. The last segment of the Storage File Share resource manager ID should be `/fileshares/` not `/shares/` in Swagger since the backend service is using `/fileshares/`. See more details from https://github.com/Azure/azure-rest-api-specs/issues/25209. So the feature of `TransportMount` isn't supported by TF for now due to this service API bug.

Reviewer: ```suggestion
Due to [a bug in the Azure API](https://github.com/Azure/azure-rest-api-specs/issues/25209) where the Storage File Share Id is not defined correctly, it is not currently possible to support using Transport Mount.
```

Author: Updated

---
subcategory: "Workloads"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_workloads_sap_three_tier_virtual_instance"
description: |-

>   Manages an SAP Three Tier Virtual Instance with new SAP System.

Author: Updated

---

# azurerm_workloads_sap_three_tier_virtual_instance

> Manages an SAP Three Tier Virtual Instance with new SAP System.

Author: Updated

* `environment` - (Required) The environment type for the SAP Three Tier Virtual Instance. Possible values are `NonProd` and `Prod`. Changing this forces a new resource to be created.

* `sap_fqdn` - (Required) The FQDN of the SAP system. Changing this forces a new resource to be created.

* `sap_product` - (Required) The SAP Product type for the SAP Three Tier Virtual Instance. Possible values are `ECC`, `Other` and `S4HANA`. Changing this forces a new resource to be created.

* `three_tier_configuration` - (Required) A `three_tier_configuration` block as defined below. Changing this forces a new resource to be created.

* `identity` - (Optional) An `identity` block as defined below.

* `managed_resource_group_name` - (Optional) The name of the managed Resource Group for the SAP Three Tier Virtual Instance. Changing this forces a new resource to be created.

* `tags` - (Optional) A mapping of tags which should be assigned to the SAP Three Tier Virtual Instance.

---

A `deployer_virtual_machine_packages` block supports the following:

* `storage_account_id` - (Required) The ID of the deployer VM packages storage account. Changing this forces a new resource to be created.

> * `url` - (Required) The URL of the deployer VM packages file. Changing this forces a new resource to be created.

Author: Updated

* `central_server_configuration` - (Required) A `central_server_configuration` block as defined below. Changing this forces a new resource to be created.

* `database_server_configuration` - (Required) A `database_server_configuration` block as defined below. Changing this forces a new resource to be created.

* `resource_names` - (Optional) A `resource_names` block as defined below. Changing this forces a new resource to be created.

* `high_availability_type` - (Optional) The high availability type for the three tier configuration. Possible values are `AvailabilitySet` and `AvailabilityZone`. Changing this forces a new resource to be created.

* `secondary_ip_enabled` - (Optional) Specifies whether a secondary IP address should be added to the network interface on all VMs of the SAP system being deployed. Defaults to `false`. Changing this forces a new resource to be created.

* `transport_create_and_mount` - (Optional) A `transport_create_and_mount` block as defined below. Changing this forces a new resource to be created.

* `transport_mount` - (Optional) A `transport_mount` block as defined below. Changing this forces a new resource to be created.

> ~> **Note:** The `Skip` configuration type would be enabled when the `transport_create_and_mount` and the `transport_mount` aren't set.

Author: Updated
